compendium/DeclarativeServices/src/ComponentContextImpl.cpp

```diff
@@ -229,6 +238,9 @@ void ComponentContextImpl::RemoveFromBoundServicesCache(
     bc.GetServiceObjects(ServiceReferenceU(sRef));
 
   const auto removedService = sObjs.GetService();
+  if (!removedService) {
+      return false;
+  }
```

Comment:
This will leave a number of stale bound services in the boundServicesCache. How will this affect DS?

Reply:
I realized that removedService not having a valid value can't happen here. The BoundServicesCache holds a copy of the InterfaceMapConstPtr which is a shared_ptr<InterfaceMap>. While we're holding this shared_ptr in our BoundServicesCache, the ServiceRegistrationBasePrivate object can't be destroyed. So in this case the sObjs.GetService will always return a value. I removed this test.